s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/SourceRecordIterator.java

```diff
@@ -164,6 +164,13 @@ private List<S3SourceRecord> readNext() {
                 return sourceRecords;
             }
 
+            // For bytes transformation, read whole file as 1 record
+            private boolean checkBytesTransformation(final Transformer transformer,
+                    final long numberOfRecsAlreadyProcessed) {
+                return transformer instanceof ByteArrayTransformer
+                        && numberOfRecsAlreadyProcessed == BYTES_TRANSFORMATION_NUM_OF_RECS;
```

Comment:
```suggestion
                        && numberOfRecsAlreadyProcessed > 0;
```
I was thinking a little further I think this might be a little more forgiving, in case we hit an edge case where it is incremented above 1, wdyt?

Reply:
So at every eof, readNext would be called which has to go through that check. I do not see any other path for this to hit edge cases. 
Apart from that, with this BYTES_TRANSFORMATION_NUM_OF_RECS, we are emphasising to exit after 1st rec.

Probably with that 4096 bug fix, I will remove this check too.

Reply:
If its going to be removed I am ok with that.